syft/source/source.go

```diff
@@ -25,7 +23,7 @@ type Source struct {
 type sourceDetector func(string) (image.Source, string, error)
 
 // New produces a Source based on userInput like dir: or image:tag
-func New(userInput string) (Source, func(), error) {
+func New(userInput string, registryOptions *image.RegistryOptions) (Source, func(), error) {
```

Comment:
In the future, it'd be nice to be able to abstract this, such that other consumers (such as in the changes below) don't need to worry about answering for this parameter if they don't care about registry concerns. I have no useful ideas right now though!

Reply:
I agree, and also came up blank here (without resorting to injecting not through the constructor or odd `init` voodoo)